doc/platform_api/CMIS_Diagnostic_Monitoring_Overview_in_SONiC.md

```diff
@@ -0,0 +1,1685 @@
+    8. If the transceiver supports VDM monitoring, perform the following steps:
+        1. Freeze the statistics by calling the CMIS API (`freeze_vdm_stats`) and wait for `FreezeDone` by calling `get_vdm_freeze_status`. Once the statistics are frozen, record the timestamp and copy the VDM and PM statistics from the transceiver.
+        2. Unfreeze the statistics by calling the CMIS API (`unfreeze_vdm_stats`).
+        3. Update the `TRANSCEIVER_VDM_CURRENT_SAMPLE` and `TRANSCEIVER_PM` tables with both basic and statistic instance's data read for VDM and PM.
+        4. Analyze the VDM flags by comparing the current data with the previous data and update the VDM flag, change count and time related tables.
+
+#### 4.2.2 Diagnostic Information Update During Link Down Event
+
+When a link down event is detected by the `DomInfoUpdateTask` thread, a specific subset of the diagnostic information fields are updated in the `redis-db`.
+Also, the relevant flags, change count and their set/clear time will be updated in the `redis-db`.
+
+#### 4.2.3 Details of Flag Analysis of Tables
+
+**Note**: For simplicity, this section uses DOM as an example. However, the same analysis is applicable for VDM as well.
+
+**Purpose of Flag Analysis:**
+
+The purpose of flag analysis is to track the status of various parameters and to count the number of times each DOM flag has changed. It also records the timestamp when each DOM flag was set and cleared.
+
+**Tables Used for Flag Analysis:**
+
+- `TRANSCEIVER_DOM_FLAG`: This table stores flags indicating the status of various DOM parameters.
+- `TRANSCEIVER_DOM_FLAG_CHANGE_COUNT`: This table keeps a count of how many times each DOM flag has changed.
+- `TRANSCEIVER_DOM_FLAG_SET_TIME`: This table records the timestamp when each DOM flag was set.
+- `TRANSCEIVER_DOM_FLAG_CLEAR_TIME`: This table records the timestamp when each DOM flag was cleared.
+
+**Example of Table Updates:**
+
+- **TRANSCEIVER_DOM_FLAG_CHANGE_COUNT:**
+  - Each time a flag in the `TRANSCEIVER_DOM_FLAG` table changes (either set or cleared), the corresponding count in this table is incremented.
+- **TRANSCEIVER_DOM_FLAG_SET_TIME:**
+  - When a flag is set in the `TRANSCEIVER_DOM_FLAG` table, the current timestamp (based on local timezone) is recorded in this table.
+- **TRANSCEIVER_DOM_FLAG_CLEAR_TIME:**
+  - When a flag is cleared in the `TRANSCEIVER_DOM_FLAG` table, the current timestamp (based on local timezone) is recorded in this table.
+
+#### 4.2.4 Flag Change Count and Time Set/Clear Behavior During `xcvrd` Restart
+
+During `xcvrd` stop, the `TRANSCEIVER_DOM_FLAG_CHANGE_COUNT`, `TRANSCEIVER_DOM_FLAG_SET_TIME`, and `TRANSCEIVER_DOM_FLAG_CLEAR_TIME` tables are deleted by the `xcvrd` process. When `xcvrd` is restarted, the `TRANSCEIVER_DOM_FLAG_CHANGE_COUNT`, `TRANSCEIVER_DOM_FLAG_SET_TIME`, and `TRANSCEIVER_DOM_FLAG_CLEAR_TIME` tables are recreated and the flag change count and time set/clear behavior are reset to zero.
```

Comment:
change count could be reset to 0. Time, set/clear should be NA, until the next time polling, one flag is set or cleared, these fields are updated with the actual time in string format.